src/python/WMComponent/AgentStatusWatcher/DrainStatusPoller.py

```diff
@@ -54,3 +68,46 @@ def getDrainInfo(cls):
         Return drainStats class variable
         """
         return cls.drainStats
+
+    def checkSpeedDrainThresholds(self, speedDrainConfig):
+        """
+
+        """
+        validKeys = ['CondorPriority', 'NoJobRetries', 'EnableAllSites']
+        updateConfig = False
+        # get condor jobs
+        jobs = self.condorAPI.getCondorJobs("", [])
+        if jobs is None:
+            logging.warning("There was an error querying the schedd.  Not checking speed drain thresholds.")
+            return
+
+        for k, v in speedDrainConfig.items():
+            # make sure keys are valid
+            if k in validKeys and isinstance(v['Threshold'], int) and isinstance(v['Enabled'], bool):
+                # and the drain config change isn't active
+                if not v['Enabled']:
+                    logging.info("Checking threshold for %s", k)
+                    # if number of condor jobs is below the threshold, activate speed drain config
+                    if len(jobs) < v['Threshold']:
+                        updateConfig = True
+                        if k == 'CondorPriority':
```

Comment:
A minor change, which might make this code more readable. How about setting these actions after this logic that checks whether it is enabled or not and whether it has to be enabled.

Something like after the `for` loop, we could
```
if self.speedDrainConfig['CondorPriority']['Enabled']:
  res = self.condorAPI.editCondorJobs("JobStatus=?=1 && (CMS_JobType =?= \"Production\" || CMS_JobType =?= \"Processing\")", "JobPrio", "999999"):
  if res:
    logging.info("Threshold hit for %s, Updated condor priority to 999999 for all jobs.", k)
  else:
    blah
if self.speedDrainConfig['NoJobRetries']['Enabled']:
```